spec/facter/resolvers/networking_linux_spec.rb

```diff
@@ -36,6 +36,9 @@ def ifaddr_obj(name, ip, mac, netmask, ipv4_type)
         .with('ip route show', logger: log_spy).and_return(load_fixture('ip_route_show').read)
       allow(Facter::Core::Execution).to receive(:execute)
         .with('ip -6 route show', logger: log_spy).and_return(load_fixture('ip_-6_route_show').read)
+      allow(Facter::Core::Execution).to receive(:execute)
+        .with('ip -json -6 route show default', logger: log_spy)
+        .and_return(load_fixture('ip_-json_-6_route_show_default').read)
```

Comment:
You also need to stub the Facter::Core::Execution.which('ip') call.
Also you need to set the @log inside the PrimaryInterface class. If you don't, an actual logger is instantiated and the stub for Facter::Core::Execution won't work.  Use something like this: `Facter::Util::Resolvers::Networking::PrimaryInterface.instance_variable_set(:@log, log_spy)`.